### ISSUE TITLE
Question: shared encoder receives two updates

### ISSUE THREAD

**Issue by mdiephuis** [computational-cell-analytics/micro-sam#1331]
There is something I don't fully understand, the jointSamTrainer, 

https://github.com/computational-cell-analytics/micro-sam/blob/a8a00495fc3cf6710a52c624aecbde55145daab0/micro_sam/training/joint_sam_trainer.py#L99

gets two optimization steps:
- interactive forward/backward optim step
- clears grad
- AIS forward/backward optim step

The encoder is shared, so it gets 2 updates. Is there any reason to not simply do something like:
``` 
    total = interactive_loss + ais_loss
    total.backward()
    optimizer.step()
```
Thanks!

**Comment by anwai98** [comment]
Hi @mdiephuis,

I faintly remember two reasons at the time of development why I didn't go forward with this:
1. Avoiding keeping a huge graph in memory (the interactive and automatic combined were quite expensive together)
2. Avoid doing fancy stuff with losses (I tried this by shrinking the memory footprint, but the gradient magnitudes were too different to bring both interactive and automatic loss updates to the same space to avoid dominance of one over the other -- we disliked the idea in general)

On a higher-level, we wanted to keep the intuition as expected and simple -- do interactive task first, and in the same round automatic task next.
